WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyRingViewModel.cs

```diff
@@ -74,82 +72,61 @@ protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disp
 						.PrivacyProgressUpdateTrigger
 						.CombineLatest(sizeTrigger)
 						.ObserveOn(RxApp.MainThreadScheduler)
-						.Subscribe(x =>
-						{
-							var usableHeight = x.Second.Item2;
-							var usableWidth = x.Second.Item1;
-							Margin = new Thickness(usableWidth / 2, usableHeight / 2, 0, 0);
-							NegativeMargin = new Thickness(Margin.Left * -1, Margin.Top * -1, 0, 0);
-							RefreshCoinsList(itemsSourceList);
-						})
+						.Subscribe(_ => RenderRing(itemsSourceList))
 						.DisposeWith(disposables);
 	}
 
-	private void RefreshCoinsList(SourceList<PrivacyRingItemViewModel> itemsSourceList)
+	private void RenderRing(SourceList<PrivacyRingItemViewModel> list)
 	{
-		var pockets = _walletViewModel.Wallet.GetPockets();
-		itemsSourceList.Edit(list => CreateSegments(pockets, list));
+		SetMargins();
+
+		Items.Clear();
+		list.Edit(list => CreateSegments(list));
+
+		PreviewItems.RemoveRange(1, PreviewItems.Count - 1);
```

Comment:
I would prefer this as it seems to be safer and indicated that there can be other types in the list.
```suggestion
		PreviewItems.RemoveMany(PreviewItems.OfType<PrivacyRingItemViewModel>());
```